src/main/java/com/minecraftabnormals/upgrade_aquatic/common/entities/jellyfish/helper/JellyfishSizeMap.java

```diff
@@ -0,0 +1,117 @@
+package com.minecraftabnormals.upgrade_aquatic.common.entities.jellyfish.helper;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class JellyfishSizeMap {
+    protected final TreeMap<Float, Integer> sizeMap;
+    protected final int weightTotal;
+
+    public JellyfishSizeMap(TreeMap<Float, Integer> map) {
+        this.sizeMap = map;
+        int weightTotal = 0;
+        for (int weight : this.sizeMap.values())
+            weightTotal += weight;
+        this.weightTotal = weightTotal;
+    }
+
+    public int size() {
+        return sizeMap.size();
+    }
+
+    public int totalWeight() {
+        return this.weightTotal;
+    }
+
+    public Float randomSize(Random rand) throws RuntimeException {
```

Comment:
Why does this return the float wrapper?

Reply:
I think it was a holdover from when it was initialised as null. I've changed it now.